src/video/wayland/SDL_waylandwindow.c

```diff
@@ -393,7 +376,7 @@ SetFullscreen(SDL_Window *window, struct wl_output *output, SDL_bool commit)
     /* The desktop may try to enforce min/max sizes here, so turn them off for
      * fullscreen and on (if applicable) for windowed
      */
-    SetMinMaxDimensions(window, SDL_FALSE);
+    SetMinMaxDimensions(window, SDL_TRUE);
```

Comment:
This results in a double-commit, as the commit is supposed to happen further down in the function alongside the fullscreen changes. This system was here to fix the very same thing this patchset supposedly fixes... so what regressed here?

Reply:
Not sure, but without that extra commit, going from windowed to fullscreen on GNOME doesn't work properly.  If it's a compositor bug, probably better to revert this and file an upstream bug.

Dropped for now.

Reply:
I tried 2.0.20 and it worked, so I did some bisecting to find the commit that broke it.

68a71f91c6cbb1458dc95e2d994d4522afd92c5e is the culprit.

Reply:
Odd, that commit is the one thing that fixed startup issues (like the ones described in #5257). The weirder part is that tests start and end on GNOME with KDE as a secondary test, so I'm surprised that's the busted one rather than KWin.

Reply:
Scratch that, for some reason it worked a few times when I ran it in the debugger, then it stopped again with no obvious reason why, so my previous hypothesis was wrong.

The culprit seems to be calling xdg_toplevel_set_min_size()/xdg_toplevel_set_max_size() before committing the fullscreen change.  If these specific calls are not used just before the fullscreen setting committed, everything works.